### PR TITLE
ISPN-5356 Honour FAIL_SILENTLY with Optimistic caches

### PR DESCRIPTION
**Pull request by galderz**
https://issues.jboss.org/browse/ISPN-5356

### CODE REVIEW

**Comment by pruivo** [comment]
a couple of comments:
first, I don't know why do we support ```FAIL_SILENTLY``` in transactions. The data will become inconsistency (use putAll instead of transaction will achieve the same thing. With write-skew check it will become worse...).
second, your changes does not work in transactions with multiple keys.

**Comment by danberindei** [comment]
@pruivo +1, we shouldn't support this for all transactions. If an implicit transaction created for a `cache.put(k, v)` fails, it should be `CacheImpl.executeCommandAndCommitIfNeeded` that catches the exception and ignores it.

File: core/src/test/java/org/infinispan/lock/OptimisticAPITest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        assertTrue(prepare());
+
+        tm.resume(t2);
+        assertTrue(prepare());
+
+        tm.resume(t1);
+        commit();
+
+        tm.resume(t2);
+        commit();
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/lock/OptimisticAPITest.java]: Should also test that the second transaction doesn't overwrite the first transaction's value.

**Comment by galderz** [comment]
Closing, submitting a new PR with a javadoc update.

**Comment by galderz** [comment]
See #3431

**Pull request closed without merging**